### PR TITLE
[FSSDK-8581] chore: Prepare for 4.0.0 release

### PR DESCRIPTION
**Pull request by mikechu-optimizely**
## Summary
- Incremement version numbers removing the beta tag
- Update changelog, readme.md, & license

## Test plan
- All unit and e2e test suites should pass

## Issues
- [FSSDK-8581](https://jira.sso.episerver.net/browse/FSSDK-8581)

### CODE REVIEW

**Review by muzahidul-opti — APPROVED** [review]
LGTM. Just a quick thought, shouldn't we need to update the readme file?

**Comment by mikechu-optimizely** [comment]
> LGTM. Just a quick thought, shouldn't we need to update the readme file?

That's a good question. What are you and @jaeopt adding/updating to the README.md?

**Comment by muzahidul-opti** [comment]
> > LGTM. Just a quick thought, shouldn't we need to update the readme file?
> 
> That's a good question. What are you and @jaeopt adding/updating to the README.md?

> For mobile suite version number. For the 

> > LGTM. Just a quick thought, shouldn't we need to update the readme file?
> 
> That's a good question. What are you and @jaeopt adding/updating to the README.md?

> For mobile suit, sample API and version need to be updated.

**Comment by muzahidul-opti** [comment]
> > LGTM. Just a quick thought, shouldn't we need to update the readme file?
> 
> That's a good question. What are you and @jaeopt adding/updating to the README.md?

> For mobile suit, sample api and version need to be updated.

**Comment by mikechu-optimizely** [comment]
Ah yes. I see you have a version number in the Swift SDK README.md then searching I see that the C# SDK has a version number that's not been updated since 3.2 😬 .  

I'll update that and the SDK Deployment runbook.  Thanks.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,31 @@
 # Optimizely C# SDK Changelog
 
+## 4.0.0
+January 16th, 2024
+
+### New Features
+- Promote Advanced Audience Targeting to General Availability
```
Comment (jaeopt) [inline CHANGELOG.md:7]: I think this is fine with CHANGELOG, but we need the full description (copy from 4.0.0-beta) for ODP feature in release note.

**Review by jaeopt — APPROVED** [review]
LGTM

**Comment by jaeopt** [comment]
> > LGTM. Just a quick thought, shouldn't we need to update the readme file?
> 
> That's a good question. What are you and @jaeopt adding/updating to the README.md?

We can consider adding a line what his major release about. They can look at the release note, but one line of summary will be helpful.

**Comment by mikechu-optimizely** [comment]
> > That's a good question. What are you and @jaeopt adding/updating to the README.md?
> 
> We can consider adding a line what his major release about. They can look at the release note, but one line of summary will be helpful.

What's the copy for the README one-liner addition?  Will it age well with the 4.x.x?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,75 @@
...
+## 4.0.0
+January 16th, 2024
+
+### New Features
+
+#### Polling warning
+
+Add warning to polling intervals below 30 seconds ([#365](https://github.com/optimizely/csharp-sdk/pull/365))
+
+#### Advanced Audience Targeting
```
Comment (jaeopt) [inline CHANGELOG.md:12]: Can we switch these? ODP first :)

**Review by jaeopt — APPROVED** [review]
LGTM - nit

**Pull request merged**